### PR TITLE
SALTO-5692, SALTO-6291: Okta: use `Nock` for deploy tests

### PR DESCRIPTION
**Pull request by Nurdok**
This PR:
- Replaces the HTTP-recording-based deploy tests in Okta to use the `nock` library (main file to review is `adapter.test.ts`)
- Adds a Python utility script to capture HTTP calls from Salto logs in `nock`'s format (next to the existing script for axios). The script also supports converting existing `json` files to `nock`'s format, to ease migrations of existing tests to `nock`.

Note that fetch-tests are still using `mock-axios-adapter`.

---

_Additional context for reviewer_:
None

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/adapter-components/scripts/client/mock_replies_nock.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    if not http_data["path"].startswith("/"):
+        parsed = urllib.parse.urlparse(http_data["path"])
+        http_data["path"] = parsed.path
+        http_data["scope"] = f"{parsed.scheme}://{parsed.netloc}"
+    if "data" in http_data:
+        http_data.setdefault("body", http_data.pop("data"))
+    if "body" in http_data and http_data["body"] is None:
+        del http_data["body"]
+    http_data.setdefault("reqHeaders", http_data.pop("headers", None))
+    if "queryParams" in http_data:
```
Comment (shir-reifenberg) [inline packages/adapter-components/scripts/client/mock_replies_nock.py:45]: as this is generic, should we also take into account `headers` ?

**Review by shir-reifenberg — APPROVED** [review]
Looks great!!! 👑  and will be very helpful
a few more comments -
1. Can you please create a dedicated ticket explaining the issue and describing the research you did? so it will be easier to track what and why we did it 
2. I think it's worth adding some kind of readme file (maybe to okta's mock replies folder) to that will describe how to set up a new test for a new type

File: packages/okta-adapter/test/adapter.test.ts [hunk 2]
```diff
@@ -518,11 +537,13 @@ describe('adapter', () => {
...
         expect(result.appliedChanges).toHaveLength(1)
         expect(getChangeData(result.appliedChanges[0] as Change<InstanceElement>).value.id).toEqual('group-fakeid1')
+        expect(nock.pendingMocks()).toHaveLength(0)
       })
 
       it('should successfully modify a group', async () => {
+        loadMockReplies('group_modify.json')
         const updatedGroup1 = group1.clone()
-        updatedGroup1.value.name = 'Programmers'
+        updatedGroup1.value.profile.name = 'Programmers'
```
[the remark this answers is not in the document]
Comment (Nurdok) [reply packages/okta-adapter/test/adapter.test.ts:546]: Yep, it's great that this change actually found some bugs (this one is just in the test, but in the Application PR it found a bunch of real bugs in my migration).

File: packages/okta-adapter/test/adapter.test.ts [hunk 3]
```diff
@@ -637,17 +668,15 @@ describe('adapter', () => {
...
-        expect(mockAxiosAdapter.history.post.length).toBe(1)
-        const addUserReq = mockAxiosAdapter.history.post[0]
-        expect(addUserReq.url).toEqual('/api/v1/users')
-        expect(addUserReq.params).toEqual({ activate: 'false' })
 
         expect(result.errors).toHaveLength(0)
         expect(result.appliedChanges).toHaveLength(1)
         expect(getChangeData(result.appliedChanges[0]).elemID.getFullName()).toEqual('okta.User.instance.user1')
         expect(getChangeData(result.appliedChanges[0] as Change<InstanceElement>).value.id).toEqual('fakeid123')
+        expect(nock.pendingMocks()).toHaveLength(0)
```
[the remark this answers is not in the document]
Comment (Nurdok) [reply packages/okta-adapter/test/adapter.test.ts:676]: Yes, `Nock` matches on all params including query and body.

File: packages/adapter-components/scripts/client/mock_replies_nock.py (hunk 1 above)
Comment (Nurdok) [reply packages/adapter-components/scripts/client/mock_replies_nock.py:45]: Headers are handled on line 44.

File: packages/adapter-components/scripts/client/mock_replies_nock.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+
+def read_http_data_from_salto_log(logpath: str) -> list[dict[str, Any]]:
+    """Return HTTP response data from a Salto log file."""
+    with open(logpath) as logfile:
+        return [
+            json.loads(l[l.find(": ") + 2 :].strip())
+            for l in logfile.readlines()
+            if "Full HTTP response" in l
+        ]
```
[the remark this answers is not in the document]
Comment (Nurdok) [reply packages/adapter-components/scripts/client/mock_replies_nock.py:30]: There's some setup needed to be able to `import` Python scripts from one another (setting `PYTHONPATH` when running and adding `__init__.py` files to the directory tree). For a 5-line function I'd say this isn't worth the complication.

**Comment by Nurdok** [comment]
Opened a ticket and added a README :)

File: packages/okta-adapter/test/mock_replies/brand_add.json
[the remark this answers is not in the document]
Comment (Nurdok) [reply packages/okta-adapter/test/mock_replies/brand_add.json:1]: Ignoring for now, let's see if this becomes a problem :)

**Pull request merged**